db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/PredicateList.java

```diff
@@ -3639,4 +3640,37 @@ public boolean isUnsatisfiable() {
                 return true;
         return false;
     }
+    @Override
+    public boolean canSupportIndexExcludedNulls(int tableNumber, ConglomerateDescriptor cd, TableDescriptor td) throws StandardException {
+        return exclusionSupport(tableNumber,cd,td,true);
+    }
+
+    @Override
+    public boolean canSupportIndexExcludedDefaults(int tableNumber, ConglomerateDescriptor cd, TableDescriptor td) throws StandardException {
+        return exclusionSupport(tableNumber,cd,td,false);
+    }
+
+    private boolean exclusionSupport(int tableNumber, ConglomerateDescriptor cd, TableDescriptor td, boolean excludeNulls) throws StandardException {
+        int size = size();
+        if (size == 0)
+            return false;
+        int[] baseColumnPositions = cd.getIndexDescriptor().baseColumnPositions();
+        ExcludedIndexColumnVisitor excludedIndexColumnVisitor = new ExcludedIndexColumnVisitor(0,baseColumnPositions[0],excludeNulls,excludeNulls?null:
+                td.getDefaultValue(cd.getIndexDescriptor().baseColumnPositions()[0]));
+        boolean canSupportExcludedColumns = false;
+        for(int index=0;index<size;index++){
+            Predicate pred=elementAt(index);
+            if(pred.isOrList()){
+                return false;
```

Comment:
I was not going to perform or validation for this first release.  If you have or predicates we will not use the index.  It is a bit restrictive but handles our customers use case.